### DISCUSSION TITLE
Smart contract best practice when creating season based collection

### DISCUSSION THREAD

**Discussion by dcryptodev** [hashlips-lab/nft-erc721-collection#212] [General]
Hi all,

Im currently working on building a seasons based NFT collection where we would launch a new season NFT's every 3/4 months. 

I was wondering if anyone know the best practices associated with this from a smart contract perspective?

Would it be a best practice to create different smart contracts for each session? i.e:
- SEASON 1 -  Smart Contract A
- SEASON 2 -  Smart Contract B
- SEASON 3 -  Smart Contract C

I was thinking this would make sense since I want the seasons to have different collections on Opensea.
Note: each session would have approx ~5000-7000 NFTs ( this will be on the Polygon blockchain). 

Just wanted to ask anyone if this makes sense?

**Comment by ecerroni** [comment]
@dcryptodev it makes sense because of the things you listed:
- it is seasonal
- each season will be a different collection on opensea
- each season has around 5k tokens

Doing everything using just one contract would end having one huge collection with around 20k tokens that would not be too appealing or easy to manage, I guess :)

**Reply by dcryptodev** [reply]
Thank you so much for your feedback! +1
